STUDIO/org.openl.rules.webstudio/src/org/openl/rules/rest/RestExceptionMapper.java

```diff
@@ -1,18 +1,81 @@
 package org.openl.rules.rest;
 
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
 import javax.ws.rs.core.Response;
 import javax.ws.rs.ext.ExceptionMapper;
 
+import org.openl.rules.rest.exception.ValidationException;
+import org.openl.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
 public class RestExceptionMapper implements ExceptionMapper<Exception> {
 
+    private static final Logger LOG = LoggerFactory.getLogger(RestExceptionMapper.class);
+
     @Override
     public Response toResponse(Exception e) {
-        if (e instanceof SecurityException) {
-            return  Response.status(Response.Status.FORBIDDEN.getStatusCode()).entity("You haven't privileges to do that.").build();
-        } else if (e instanceof IllegalArgumentException) {
-            return Response.status(Response.Status.CONFLICT.getStatusCode()).entity(e.getMessage()).build();
+        ResponseStatus status = e.getClass().getAnnotation(ResponseStatus.class);
+        if (status != null) {
+            HttpStatus code = status.code();
+            if (e instanceof ValidationException) {
+                return handleBindingResult(code, ((ValidationException) e).getBindingResult());
+            } else {
+                return Response.status(code.value()).entity(mapCommonException(code, e)).build();
+            }
+        } else {
+            HttpStatus code = HttpStatus.INTERNAL_SERVER_ERROR;
+            LOG.error(e.getMessage(), e);
+            return Response.status(code.value()).entity(mapCommonException(code, e)).build();

```

Comment:
Does this change affect error handling for jsf errors?
For the code:
`jsf.ajax.addOnError(function (data) {
  var errorMessage = data.errorMessage
} `

errorMessage is text or json?

Reply:
RestExceptionMapper are applied only on ```/rest/*``` urls, and it doesn't affect old validation approach